File: py/linear_search.py
def linear_search(list, target):
    """
    Returns the index position of the element if found, else returns None
    """

    for i in range(0, len(list)):
        if list[i] == target:
            return i
    return None

def verify(index):
    if index is not None:
        print("Target found at index: " + str(index))
    else:
        print("Target not found in list")


if __name__ == '__main__':
    # value not in list
    numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
    result = linear_search(numbers, 12)
    verify(result)

    # value at index 5
    result_2 = linear_search(numbers, 6)
    verify(result_2)

File: javascript/linear_search.js
const linear_search = (search_array, target) => {
    for (let i = 0; i < search_array.length; i++) {
        if (search_array[i] === target) {
            return i;
        }
    }

    return null;
}

const verify = (index) => {
    if (index) {
        console.log('Target found at index: ' + index.toString())
    } else {
        console.log('Target not found')
    }
}


const main = () => {
    const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

    const result = linear_search(numbers, 12)
    verify(result)

    const result_2 = linear_search(numbers, 6)
    verify(result_2)
}

main()